library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/ResourceLoader.java

```diff
@@ -50,6 +51,19 @@ public interface ResourceLoader {
 		return ResourceLoaderImpl.get(type);
 	}
 
+	/**
+	 * Gets the static resource manager for a given type.<br>
+	 * Static resources are different from regular resources in that they
+	 * are available as soon as the manager is requested and are (theoretically) immutable.<br>
+	 * It is <i>highly</i> recommended for mods to personally cache data they get from the provided manager in order to
+	 * prevent changes to userspace packs modifying data at runtime.
+	 * @param type the given resource type
+	 * @return the static resource manager instance
+	 */
+	static @NotNull MultiPackResourceManager getStaticResourceManager(@NotNull ResourceType type){
```

Comment:
Any reason to return `MultiPackResourceManager` and not just `ResourceManager`?
So far `ResourceManager` is usually used for APIs that provide a resource manager.

Also, given the API structure, this should not be `static` but be something to implement directly on the `ResourceLoaderImpl` class like every other method present in this interface.

Reply:
The former I agree with and will be changed accordingly in the next commit, but the latter I have to disagree with—the creation and storage of the `StaticResourceManager`s that back the Static Resources API are both entirely static-scoped, so it's both less verbose and more aligned with the data structure of the API itself to get the static resource managers in a static method.

If this is a sticking point I can change it, but I feel like it makes more sense this way.

Reply:
I'm inclined to agree that this should be static, given that a static resource manager is not tied to a `ResourceLoader` in any way